src/matched-data.spec.ts

```diff
@@ -62,6 +62,41 @@ it('does not include valid data from invalid oneOf() chain group', done => {
   });
 });
 
+describe('with chains that contain .if()', () => {
+  it('does not include data of chain with early failed condition', done => {
+    const req = {
+      query: { foo: 'foo' },
+    };
+
+    const middleware = check('foo')
+      .if((value: any) => /\d/.test(value))
+      .equals('foo');
+
+    middleware(req, {}, () => {
+      expect(matchedData(req)).toEqual({});
+      done();
+    });
+  });
+
+  it('includes data of chain with late failed condition', done => {
+    const req = {
+      query: { foo: 'foo' },
+    };
+
+    const middleware = check('foo')
+      .equals('foo')
+      .if((value: any) => /\d/.test(value))
```

Comment:
This could use something that would actually work if used in production, `/\d/` will never pass as `foo` doesn't have any digits.